tests/util/FitsComparatorTest.cpp

```diff
@@ -0,0 +1,81 @@
+#include <boost/test/unit_test.hpp>
+#include <core/Exceptions.h>
+#include <util/FitsComparator.h>
+#include <core/IIPBase.h>
+#include <fitsio.h>
+#include <map>
+#include <iostream>
+
+namespace ut = boost::unit_test;
+
+struct FitsComparatorFixture : IIPBase {
+    unique_ptr<FitsComparator> _fc;
+    unique_ptr<FitsComparator> _fc2;
+    string _file1;
+    string _file2;
+    string _log_dir;
+    fitsfile* _fptr;
+    int _status = 0;
+    FitsComparatorFixture() : IIPBase("ForwarderCfg.yaml", "FitsComparatorTest"){
+        BOOST_TEST_MESSAGE("Setup FitsComparator test");
+        _log_dir = _config_root["LOGGING_DIR"].as<std::string>();
+        BOOST_TEST_REQUIRE(ut::framework::master_test_suite().argc == 3);
+        BOOST_TEST(ut::framework::master_test_suite().argv[1] == "--data");
+        string path(ut::framework::master_test_suite().argv[2]);
+        _file1 = path + "/test1.fits";
+        _file2 = path + "/test2.fits";
+        fits_open_file(&_fptr, _file1.c_str(), READONLY, &_status);
+        _fc = unique_ptr<FitsComparator>(new FitsComparator(_file1, _file2));
+        _fc2 = unique_ptr<FitsComparator>(new FitsComparator(path + "/test3.fits", path + "/test4.fits"));
+    }
+    ~FitsComparatorFixture(){
+        BOOST_TEST_MESSAGE("Teardown FitsComparatorTest fixture");
+        fits_close_file(_fptr, &_status);
+    }
+};
+
+BOOST_FIXTURE_TEST_SUITE(FitsComparatorTest, FitsComparatorFixture);
+
+BOOST_AUTO_TEST_CASE(constructor){
+    BOOST_CHECK_NO_THROW(FitsComparator f(_file1, _file2));
+
+    BOOST_CHECK_THROW(FitsComparator f("_file1", _file2), L1::CfitsioError);
+
+    BOOST_CHECK_THROW(FitsComparator f(_file1, "file.f"), L1::CfitsioError);
+}
+
+BOOST_AUTO_TEST_CASE(compare_by_hdu){
+    BOOST_CHECK_EQUAL(_fc->compare_by_hdu(), true);
```

Comment:
What your unit test is doing is called `integration testing`, which is have an external file and compare them each other. What we would like to have is a unit test, which is `create a two fitsfile here. write some random numbers. then use that programmatically generated fitsfiles to check your code. If you do not understand this part, we should discuss.